pkg/booleanpolicy/augmentedobjs/construct.go

```diff
@@ -143,12 +143,19 @@ func ConstructDeploymentWithNetworkFlowInfo(
 }
 
 // ConstructDeployment constructs the augmented deployment object.
-func ConstructDeployment(deployment *storage.Deployment, images []*storage.Image) (*pathutil.AugmentedObj, error) {
+func ConstructDeployment(deployment *storage.Deployment, images []*storage.Image, applied *NetworkPoliciesApplied) (*pathutil.AugmentedObj, error) {
 	obj := pathutil.NewAugmentedObj(deployment)
 	if len(images) != len(deployment.GetContainers()) {
 		return nil, errors.Errorf("deployment %s/%s had %d containers, but got %d images",
 			deployment.GetNamespace(), deployment.GetName(), len(deployment.GetContainers()), len(images))
 	}
+
+	appliedPolicies := pathutil.NewAugmentedObj(applied)
```

Comment:
Is it okay passing `nil` to that function? It later does `reflect.ValueOf(nil)` which produces `<invalid reflect.Value>`

Try: `fmt.Printf("value: %+v", reflect.ValueOf(nil))`

Reply:
TIL that the reflect package treats `nil` values differently from `nil` pointers: https://go.dev/play/p/We8LUJzQBPh

Since we're providing a nil pointer to a struct the reflected value is `<nil>`. Added a test case to show that as well as you suggested below.